applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -60,368 +57,300 @@ public ConfigurationClient(String connectionString, HttpPipeline pipeline) {
         }
 
-        return result.flatMapMany(p ->
-                Flux.just(new RestPagedResponseImpl<>(p.body().items(), p.body().nextPageLink(), p.request(),
-                        p.headers(), p.statusCode())));
+        return deleteKeyValue(key, null, null);
     }
 
     /**
-     * Gets all ConfigurationSetting settings.
+     * Deletes the ConfigurationSetting.
      *
-     * @param nextPageLink The NextLink from the previous successful call to List operation.
-     * @return the observable to the Page&lt;ConfigurationSetting&gt; object.
-     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @param key   key of the keyValue to delete
+     * @param label Optional. If not specified, {@link ConfigurationSetting#NULL_LABEL} is used.
+     * @param etag  Optional. If specified, will only delete the key if its current etag matches. (ie. No one has
+     *              changed the value yet.)
+     * @return the deleted ConfigurationSetting or none if didn't exist.
      */
-    private Flux<RestPagedResponse<ConfigurationSetting>> listNextAsync(@NonNull String nextPageLink) {
-        return listNextSinglePageAsync(nextPageLink)
-                       .concatMap(page -> {
-                           String nextPageLink1 = page.nextLink();
-                           if (nextPageLink1 == null) {
-                               return Flux.just(page);
-                           }
-                           return Flux.just(page).concatWith(p -> listNextAsync(nextPageLink1));
-                       });
-    }
-
-    private Flux<RestPagedResponse<ConfigurationSetting>> listNextSinglePageAsync(@NonNull String nextPageLink) {
-        if (nextPageLink == null) {
-            throw new IllegalArgumentException("Parameter nextPageLink is required and cannot be null.");
+    public Mono<RestResponse<ConfigurationSetting>> deleteKeyValue(String key, String label, String etag) {
+        if (key == null || key.isEmpty()) {
+            throw new IllegalArgumentException("Parameter 'key' is required and cannot be null or empty");
+        } else if (label == null) {
+            label = ConfigurationSetting.NULL_LABEL;
         }
-        return service.listKeyValuesNext(baseUri.toString(), nextPageLink)
-                       .flatMapMany(p -> Flux.just(new RestPagedResponseImpl<>(p.body().items(), p.body().nextPageLink(), p.request(), p.headers(), p.statusCode())));
+
+        return service.delete(baseUri.toString(), key, label, getETagValue(etag), null);
     }
 
     /**
      * Places a lock on ConfigurationSetting.
-     * @param key key name
-     * @return ConfigurationSetting
+     *
+     * @param key The key to lock.
+     * @return ConfigurationSetting that was locked
+     * @throws com.azure.common.http.rest.RestException with status code 404 if the {@param key} does not exist.
      */
     public Mono<RestResponse<ConfigurationSetting>> lockKeyValue(String key) {
```

Comment:
Same here and just key and optional label...

Reply:
There's an overload below that takes `lock(String key, String label)`, unfortunately "default" parameters are not a thing in java.